src/Twig/WidgetExtension.php

```diff
@@ -13,23 +14,41 @@
  */
 class WidgetExtension extends AbstractExtension
 {
+    /** @var Snippets */
+    private $snippets;
+
+    public function __construct(Snippets $snippets)
+    {
+        $this->snippets = $snippets;
+    }
     /**
      * {@inheritdoc}
      */
-    public function getFunctions()
+    public function getFunctions(): array
     {
         $safe = ['is_safe' => ['html']];
         $env = ['needs_environment' => true];
```

Comment:
why they need twig environment?

Reply:
Some widgets render twig templates (like the Newswidget), and if we pass it in, they can use the context from the "parent" template. 

(for example, if somebody wants to make a widget that does something, based on what's on the page)

Reply:
As an added bonus of 76eafe7b995b50a02a296879572cdb17ad38969e, we don't need environment here. :-)